README.md

```diff
@@ -1,105 +1,80 @@
-- Other non-epoch BEEFY-justifications can be stored only in memory (no need to persist them at all).
-- RPC side:
-  - Secondary-finalized-block-stream, something like grandpa_justifications RPC (Jon’s PR)
-  - On-demand retrieval of BEEFY-justifications from epoch-blocks stored in the DB.
-    (getBlock is enough - cause it returns all justifications)
-
-#### Brain dump for the ETH contract:
-1. Imports:
-  - `MerkleMountRangeRootHash`
-  - `CurrentBlockHash`
-  - MerkleRoot of PublicKeys of the next Authority Set
-  - Bit-vec of validators that signed (only accept if there is enough signatures)
-  - Merkle root of all signatures
-2. Starts interactive verification process
-  - Pick ⅓ validators that signed at random
-  - Request their signatures
-3. Required signatures are then submitted:
-  - We get ⅓ signatures + merkle proof that all of them were part of the initial set
-  - We get ⅓ public keys (or their hashes) + merkle proof that they are part of the current
-    Authority Set (stored in the contract)
-  - We `ecrecover` the signatures and make sure the public keys match the ones we got and the merkle
-    proof is valid.
+blockchains, currently mainly ETH mainnet.
+
+It can be thought of as an (optional) Bridge-specific Gadget to the Grandpa Finality protocol.
+The Protocol piggybacks on many assumptions provided by Grandpa, and is required to be built
+on top of it to work correctly.
+
+🚧 BEEFY is currently under construction - a hardhat is recommended beyond this point 🚧
+
+## Contents
+- [Build](#build)
+- [Documentation](#documentation)
+- [Project Layout](#project-layout)
+- [Running BEEFY](#running-beefy)
+
+## Build
+To get up and running you need both stable and nightly Rust. Rust nightly is used to build the Web
+Assembly (WASM) runtime for the node. You can configure the WASM support as so:
+
+```
+rustup install nightly
+rustup target add wasm32-unknown-unknown --toolchain nightly
+```
+
+Once this is configured you can build and test the repo as follows:
+
+```
+git clone https://https://github.com/paritytech/grandpa-bridge-gadget.git
+cd grandpa-bridge-gadget
+cargo build --all
+cargo test --all
+```
+
+If you need more information about setting up your development environment Substrate's
+[Getting Started](https://substrate.dev/docs/en/knowledgebase/getting-started/) page is a good
+resource.
+
+## Documentation
+
+The best way to get going with BEEFY is by reading the [walkthrough](./docs/walkthrough.md) document!
+This document puts BEEFY into context and provides motivation for why this project has been started.
+In addition to that the current status as well as a preliminary roadmap is presented.
+
+[beefy](./docs/beefy.md) is a collection of early notes and ideas, stil worth checking out.
+
+## Project Layout
+
+What follows is an overview of how the project repository is laid out. The main components are the
+`beefy-gadget` which is a POC of the BEEFY round logic. The BEEFY `pallet` which is mainly a thin
```

Comment:
```suggestion
`beefy-gadget` which is a POC of the BEEFY round logic. The BEEFY `pallet` which is mainly a thin
```
Might be worth either renaming the pallet folder to `pallet-beefy` maybe or `beefy-gadget` folder to simply `gadget/service`?

Reply:
I will go with `beefy-pallet`. I would like to rework/rename `node-example` to `beefy-node` in a later PR as well.